pkg/server/gitproviders/gitprovider.go

```diff
@@ -117,6 +118,32 @@ func (s *GitProviderService) GetConfigForUrl(repoUrl string) (*gitprovider.GitPr
 	return nil, errors.New("git provider not found")
 }
 
+func (s *GitProviderService) GetGitProviderForHttpRequest(req *http.Request) (gitprovider.GitProvider, error) {
+	var providerId string
+
+	gitProviders, err := s.configStore.List()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, p := range gitProviders {
+		header := req.Header.Get(config.GetWebhookEventHeaderKeyFromGitProvider(p.Id))
+		if header == "" {
+			continue
+		} else {
+			providerId = p.Id
+			break
+		}
+	}
+
+	return s.newGitProvider(&gitprovider.GitProviderConfig{
```

Comment:
Why don't we construct the entire provider (with the token, username and apiUrl)?